src/WebJobs.Extensions.DurableTask/DurableTaskExtension.cs

```diff
@@ -515,16 +509,13 @@ private void InitializeLinuxLogging()
             // Since our logging payload can be quite large, linux telemetry by default
             // disables verbose-level telemetry to avoid a performance hit.
             bool enableVerbose = this.Options.Tracing.AllowVerboseLinuxTelemetry;
-            this.eventSourceListener = new EventSourceListener(linuxLogger, enableVerbose, this.TraceHelper, this.defaultDurabilityProvider.EventSourceName);
+            this.eventSourceListener = new EventSourceListener(this.linuxLogger, enableVerbose, this.TraceHelper, this.defaultDurabilityProvider.EventSourceName);
         }
 
         /// <inheritdoc />
         public void Dispose()
         {
-            // Not flushing the linux logger may lead to lost logs
-            // 40 seconds timeout because we write normally every 30 seconds, so we're just
-            // adding an extra 10 seconds to flush.
-            LinuxAppServiceLogger.Logger?.Stop(TimeSpan.FromSeconds(40));
+            this.linuxLogger?.Dispose();

```

Comment:
Slight change here - need to put the linuxLogger as the last dispose, because you want to ditch the eventSourceListener that depends on it first.

Reply:
Ah, good catch, thanks! I have moved this down as the last step.
However, I figured this would ideally be order-agnostic, and the way to make this happen is, as I understand, to override the Dispose method of the eventSourceListener so that it would dispose of the logger as well. My last PR does this.

Let me know if that seems right to you!

Reply:
That change is absolutely fine..._but_ if you do it this way, I recommend removing the `linuxLogger` variable all together from this class - it no longer needs a direct reference at all :) It just needs to be a local passed to the event listener creation.

Reply:
Ah, of course! Removed